### PR TITLE
Apply view model pattern to local episodic effects

### PR DESCRIPTION
**Pull request by heyagosper**
refactored local episodic events code

I renamed all the components and references to xxxEvent instead of xxxEffect to match what comes through from the back end. It is only "Effect" in what the user sees. I'll change this when I change the backend later in this sprint but for now it made sense to do it this way to me.

### CODE REVIEW

File: src/client/app/shared/json-data-view-model/json-view-model.service.ts [hunk 1]
```diff
@@ -33,7 +34,9 @@ export class JsonViewModelService {
         siteLogViewModel.siteLog.gnssAntennas = this.dataToViewModel(siteLogDataModel.gnssAntennas, GnssAntennaViewModel);
         siteLogViewModel.siteLog.frequencyStandards = this.dataToViewModel(siteLogDataModel.frequencyStandards,
             FrequencyStandardViewModel);
-        siteLogViewModel.siteLog.humiditySensors = this.dataToViewModel(siteLogDataModel.humiditySensors,
+        siteLogViewModel.siteLog.localEpisodicEvents = this.dataToViewModel(siteLogDataModel.localEpisodicEventsSet,
+            LocalEpisodicEventViewModel);
+         siteLogViewModel.siteLog.humiditySensors = this.dataToViewModel(siteLogDataModel.humiditySensors,
```
Comment (oehm-smith) [inline src/client/app/shared/json-data-view-model/json-view-model.service.ts]: Indent incorrect

File: src/client/app/shared/json-data-view-model/json-view-model.service.ts [hunk 2]
```diff
@@ -86,43 +90,37 @@ export class JsonViewModelService {
...
 
     /* ***************************** Helper functions ***************************** */
-
     /**
-    * Translate data model to view model
-    * @param dataModels - array of data model items to convert
-    * @param viewModelInstance - used as a template to copy and populate.  And has methods used.
-    * @returns {AbstractViewModel[]} that is the super type of all view model types
-    */
+     * Translate data model to view model
```
Comment (oehm-smith) [inline src/client/app/shared/json-data-view-model/json-view-model.service.ts]: Why is the formatting putting in these spaces?

Comment (heyagosper) [inline src/client/app/shared/json-data-view-model/json-view-model.service.ts]: I don't know what you mean

File: src/client/app/site-info/site-info.component.ts [hunk 3]
```diff
@@ -48,14 +48,14 @@ export class SiteInfoComponent implements OnInit, OnDestroy {
     isMetaCustodianOpen: false,
     isReceiverGroupOpen: false,
     isReceiversOpen: [],
-    isEpisodicEffectGroupOpen: false,
-    isEpisodicEffectOpen: [],
+    isLocalEpisodicEventGroupOpen: false,
```
Comment (oehm-smith) [inline src/client/app/site-info/site-info.component.ts]: All this gruff isn't needed anymore.

File: src/client/app/site-info/site-info.component.ts [hunk 4]
```diff
@@ -167,7 +167,9 @@ export class SiteInfoComponent implements OnInit, OnDestroy {
 
           this.setGnssReceivers(this.siteLogModel.gnssReceivers);
           this.setSurveyedLocalTies(this.siteLogModel.surveyedLocalTies);
-          this.setEpisodicEffects(this.siteLogModel.localEpisodicEventsSet);
+          this.setLocalEpisodicEvents(this.siteLogModel.localEpisodicEvents);
```
Comment (oehm-smith) [inline src/client/app/site-info/site-info.component.ts]: Yep this file can be reduced to almost nothing now.

File: src/client/app/site-info/site-info.component.ts [hunk 5]
```diff
@@ -283,6 +285,44 @@ export class SiteInfoComponent implements OnInit, OnDestroy {
   }
 
   /**
+   * Set current and previous water vapor sensors, and their show/hide flags
+   */
+  private setWaterVaporSensors(waterVaporSensors: any) {
+    this.status.isWaterVaporSensorsOpen = [];
```
Comment (oehm-smith) [inline src/client/app/site-info/site-info.component.ts]: Isn't this PR for LocalEpisodicEvents?  Why WaterVaporSensors.  Again, not much is needed in this file any more.

**Comment by oehm-smith** [comment]
Re: my comments on 'siteInfoComponent doesn't need much in it now', we can do that as a separate PR.  Apart from some cosmetic stuff, :+1:

**Pull request merged**